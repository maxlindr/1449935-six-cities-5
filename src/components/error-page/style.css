.error-page {
  height: 100vh;
}

.error-page__message {
  position: absolute;
  left: 50%;
  top: 50%;

  margin: 0;
  padding: 0 28px;

  transform: translate(-50%, -50%);
  max-width: 640px;
  width: 100%;

  font-size: 38px;
  font-style: oblique;
  font-weight: 700;
  line-height: 1.21053;
  text-align: center;
}
