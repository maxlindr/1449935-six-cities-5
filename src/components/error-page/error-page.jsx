import React from 'react';
import PropTypes from 'prop-types';
import {PageHeader} from '../page-header/page-header';
import './style.css';
import {connect} from 'react-redux';
import {getErrorMessage} from '../../store/selectors';

const ErrorPage = (props) => {
  const {message, globalMessage} = props;

  return (
    <div className="page error-page">
      <PageHeader interactive={false}/>

      <main className="page__main">
        <h1 className="error-page__message">
          {message || globalMessage}
        </h1>
      </main>
    </div>
  );
};

ErrorPage.propTypes = {
  message: PropTypes.string,
  globalMessage: PropTypes.string,
};

const mapStateToProps = (state) => ({
  globalMessage: getErrorMessage(state)
});

export {ErrorPage};
export default connect(mapStateToProps)(ErrorPage);
