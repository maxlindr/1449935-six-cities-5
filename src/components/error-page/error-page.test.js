import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {ErrorPage} from './error-page';

const renderer = new ShallowRenderer();

const PRIORITY_MESSAGE = `I am a priority message`;
const GLOBAL_MESSAGE = `I am a global message`;

describe(`ErrorPage`, () => {
  it(`Должен отображать приоритетное сообщение`, () => {
    renderer.render(
        <ErrorPage
          message={PRIORITY_MESSAGE}
          globalMessage={GLOBAL_MESSAGE}
          close={() => {}}
        />
    );

    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it(`Должен отображать глобальное сообщение`, () => {
    renderer.render(
        <ErrorPage
          message={null}
          globalMessage={GLOBAL_MESSAGE}
          close={() => {}}
        />
    );

    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
