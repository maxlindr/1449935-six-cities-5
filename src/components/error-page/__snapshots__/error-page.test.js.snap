// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage Должен отображать глобальное сообщение 1`] = `
<div
  className="page error-page"
>
  <Memo(Connect(PageHeader)) />
  <main
    className="page__main"
  >
    <h1
      className="error-page__message"
    >
      I am a global message
    </h1>
  </main>
</div>
`;

exports[`ErrorPage Должен отображать приоритетное сообщение 1`] = `
<div
  className="page error-page"
>
  <Memo(Connect(PageHeader)) />
  <main
    className="page__main"
  >
    <h1
      className="error-page__message"
    >
      I am a priority message
    </h1>
  </main>
</div>
`;
